prepare_vmware_tests: use module_defaults (#63209)
ansible/ansible: docs/docsite/rst/dev_guide/platforms/vmware_guidelines.rst

If you need to create a new VM during your test, you can use ``test_vm1``, ``test_vm2`` or ``test_vm3``. This
way it will be automatically clean up for you.

> Avoid the common boiling plate code in your test

Reviewer: ```suggestion
Avoid the common boiler plate code in your test playbook
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Since https://github.com/ansible/ansible/pull/63209, the test suite uses `modules_defaults`. This module

Reviewer: ```suggestion
From Ansible 2.10, the test suite uses `modules_defaults`. This module
```

allow us to preinitialize the following default keys of the VMware modules:

> - hostname

Reviewer: I think the module_defaults are not limited to these 4 parameters, we can drop this line.

Author: In the test-suite context, we only initialize these 4 keys. I think we need to keep the line for clarity.

- username
- password
- validate_certs

For example, the following block:

.. code-block:: yaml

>     - name: Add a VMware vSwitchs

Reviewer: ```suggestion
    - name: Add a VMware vSwitch
```

      vmware_vswitch:
        hostname: '{{ vcenter_hostname }}'
        username: '{{ vcenter_username }}'

>         username: '{{ vcenter_username }}'

Reviewer: ```suggestion
        password: '{{ vcenter_password }}'
```

        validate_certs: 'no'
        esxi_hostname: 'esxi1'
        switch_name: "boby"
        state: present

should be simplified to just:

.. code-block:: yaml

>     - name: Add a VMware vSwitchs

Reviewer: ```suggestion
    - name: Add a VMware vSwitch
```

>       vmware_vswitch:

Reviewer: ```yaml

# example_play.yml
- hosts: localhost
  module_defaults:
    group/vmware:
       hostname: '{{ vcenter_hostname }}'
       username: '{{ vcenter_username }}'
       password: '{{ vcenter_password }}' 
  tasks:
  - vmware_guest: 
     ...
  - vmware_vswitch:
      esxi_hostname: 'esxi1'
      switch_name: "vSwitch_0001"
      state: present

Author: We don't need this part, `ansible-test` will already do it.
